### PR TITLE
Analysis plugin support

### PR DESCRIPTION
**Pull request by rpavlik**
This adds initial analysis plugin support. The plugins must be "sync", and there's no cycle detection or friendly auto-setup way of accessing inputs, but it is functional.

A standalone (non-VRPN-dependent) implementation of a One Euro filter in an analysis plugin is included.

Some server refactoring was involved, which probably resolved other issues that I will mention.

### CODE REVIEW

File: inc/osvr/Client/CreateContext.h [hunk 1]
```diff
@@ -35,12 +35,17 @@
...
+
 namespace osvr {
 namespace client {
 
     OSVR_CLIENT_EXPORT common::ClientContext *
     createContext(const char appId[], const char host[] = "localhost");
 
+    OSVR_CLIENT_EXPORT common::ClientContext *
+    createAnalysisClientContext(const char appId[], const char host[],
+                                vrpn_ConnectionPtr const& conn);
```
Comment (JeroMiya) [inline inc/osvr/Client/CreateContext.h]: What's the difference between an analysis client context and a regular client context? (sorry if this is answered in the code later, commenting as I read)

Comment (rpavlik) [inline inc/osvr/Client/CreateContext.h]: Different subclass, is created with an already-existing VRPN connection (the server connection generally). Doesn't spin on startup to wait for a connection or path tree. May in the future share resources (path tree) if it makes sense. Also will probably be able to create client interface objects (ugh that name) with relative paths soon, as a way of getting its input.

**Comment by stephenmsnow** [comment]
Great news :D

**Comment by nchalon** [comment]
Thanks !

**Comment by stephenmsnow** [comment]
Hi,

I'd like to add you to my professional network on LinkedIn.

- Stephen

Confirm that you know Stephen: https://www.linkedin.com/e/v2?e=-kldnn9-igj9clrn-1p&t=ssuw&tracking=eml-guest-invite-cta&ek=invite_guest&sharedKey=TI6y34mL&invitationId=6067262159182716928








You received an invitation to connect. LinkedIn will use your email address to make suggestions to our members in features like People You May Know. Unsubscribe here: https://www.linkedin.com/e/v2?e=-kldnn9-igj9clrn-1p&t=lun&midToken=AQFmrFVhGEd36Q&ek=invite_guest&loid=AQFUplnLsrUO9AAAAVDM-6UrU_OtzqXl1sb_t_P91ACW77k1x0r-DjRJYQF0w4FLla9Fjs0RstVUfWB7IQdFQTx-wXhhtUAe17ZSDEkNxNDUe6oxEe31V_1RYVesC2ZY-1AdyKucb4Uibc3ljGFnhq_6cZfo5TkjXIvTUNajg3mFqc4D5KERr6AR-7dIjZbaJOKx&eid=-kldnn9-igj9clrn-1p


This email was sent to reply@reply.github.com.

If you need assistance or have questions, please contact LinkedIn Customer Service: https://www.linkedin.com/e/v2?e=-kldnn9-igj9clrn-1p&a=customerServiceUrl&ek=invite_guest

&copy; 2015 LinkedIn Corporation, 2029 Stierlin Court, Mountain View CA 94043. LinkedIn and the LinkedIn logo are registered trademarks of LinkedIn.

**Comment by stephenmsnow** [comment]
Stephen Snow would like to connect on LinkedIn. How would you like to respond?

Accept: https://www.linkedin.com/e/v2?e=-kldnn9-igttzrm1-4u&a=preRegInvite&ek=first_guest_reminder_01&li=13&m=hero&ts=accept_text&sharedKey=TI6y34mL&invitationID=6067262159182716928

View Stephen Snow&#39;s profile: https://www.linkedin.com/comm/profile/view?id=AAsAABtG5F8B7_-uOfC80sK7Hdvirz3ZPd_nCbY&trk=eml-first_guest_reminder_01-hero-14-profile_text&trkEmail=eml-first_guest_reminder_01-hero-14-profile_text-null-%7Ekldnn9%7Eigttzrm1%7E4u





You received an invitation to connect. LinkedIn will use your email address to make suggestions to our members in features like People You May Know. Unsubscribe here: https://www.linkedin.com/e/v2?e=-kldnn9-igttzrm1-4u&t=lun&midToken=AQFmrFVhGEd36Q&ek=first_guest_reminder_01&li=16&m=unsub&ts=HTML&eid=-kldnn9-igttzrm1-4u&loid=AQH7OrWst0G1YwAAAVDz2bMxqNvnfTiFgIy0Uh2dGFDmJeeY-imKNKw5mCM-6DG4jakAO4ya0dL9IuZgjtH77CPnNZ0161KAHOUWwor7Qf7cBc0uSWAK-Bb0AvOBHyijnfN2LXXKmIJKKJ8I8wdXxnioMTsbTclwoOLoUhoI4EU8QcCDiF0yUQN2YXGFd38zPE-g

This email was sent to reply@reply.github.com.

If you need assistance or have questions, please contact LinkedIn Customer Service: https://www.linkedin.com/e/v2?e=-kldnn9-igttzrm1-4u&a=customerServiceUrl&ek=first_guest_reminder_01

&copy; 2015 LinkedIn Corporation, 2029 Stierlin Court, Mountain View CA 94043. LinkedIn and the LinkedIn logo are registered trademarks of LinkedIn.

**Comment by stephenmsnow** [comment]
Stephen Snow would like to connect on LinkedIn. How would you like to respond?

Accept: https://www.linkedin.com/e/v2?e=-kldnn9-ih3tw7sa-8&a=preRegInvite&ek=second_guest_reminder_01&li=12&m=hero&ts=accept_text&sharedKey=TI6y34mL&invitationID=6067262159182716928

View Stephen Snow&#39;s profile: https://www.linkedin.com/comm/profile/view?id=AAsAABtG5F8B7_-uOfC80sK7Hdvirz3ZPd_nCbY&trk=eml-second_guest_reminder_01-hero-13-profile_text&trkEmail=eml-second_guest_reminder_01-hero-13-profile_text-null-%7Ekldnn9%7Eih3tw7sa%7E8





You received an invitation to connect. LinkedIn will use your email address to make suggestions to our members in features like People You May Know. Unsubscribe here: https://www.linkedin.com/e/v2?e=-kldnn9-ih3tw7sa-8&t=lun&midToken=AQFmrFVhGEd36Q&ek=second_guest_reminder_01&li=15&m=unsub&ts=HTML&eid=-kldnn9-ih3tw7sa-8&loid=AQFj1eQ9h-cA-wAAAVEX5xKVzwk2g8lWLxKowJVdXdltlEDJJJoRduqigIz3jikLOCVRB8ch7_FGFvORQlXSNYl3mnZQ71fxeA1TJOoApAiiz3nUCEUhiXZpPm90xC1dwOw8YuKaFFoHsjSMC1RxkWjOGl0886vudCwS_oh-q2f2HpuNt_NpxeT8yb1byH6g-cBH

This email was sent to reply@reply.github.com.

If you need assistance or have questions, please contact LinkedIn Customer Service: https://www.linkedin.com/e/v2?e=-kldnn9-ih3tw7sa-8&a=customerServiceUrl&ek=second_guest_reminder_01

&copy; 2015 LinkedIn Corporation, 2029 Stierlin Court, Mountain View CA 94043. LinkedIn and the LinkedIn logo are registered trademarks of LinkedIn.

**Pull request merged**